src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java

```diff
@@ -1494,6 +1488,125 @@ void loadIntoDataStore(
+                if (task.getLayer() != null && task.getLayer().getId() == null) {
+                    addToCatalog(task);
+                }
+            }
+        }
+    }
+
+    private void replaceCoverage(
+            GridFormat format, GridCoverageReader reader, CoverageStoreInfo store, ImportTask task)
+            throws IOException {
+        // Get the imported data
+        ImportData data = task.getData();
+        if (!(data instanceof SpatialFile)) {
+            throw new IllegalArgumentException(
+                    "Only FileData are supported for replace. This is " + data);
+        }
+
+        SpatialFile fileData = (SpatialFile) data;
+        File file = fileData.getFile();
+        String newFile = file.toURI().toURL().getFile();
+
+        // When replacing, we need to make sure that the provided layer and store matches what we
+        // have in the catalog.
+        CoverageStoreInfo originalStore =
+                catalog.getCoverageStoreByName(store.getWorkspace().getName(), store.getName());
+        LayerInfo inputLayer = task.getLayer();
+        LayerInfo originalLayer = checkResourceExists(store, originalStore, inputLayer);
+        CoverageInfo originalCoverage = (CoverageInfo) originalLayer.getResource();
+
+        // Marking original names for restore
+        String name = originalCoverage.getName();
+        String nativeName = originalCoverage.getNativeName();
+
+        // Marking previous file for eventual delete
+        GridCoverage2DReader oldReader = ((GridCoverage2DReader) reader);
+        ServiceInfo readerInfo = oldReader.getInfo();
+        String oldFile = readerInfo.getSource().toURL().getFile();
+
+        // Setting up the new coverage
+        CatalogRepository repository = catalog.getResourcePool().getRepository();
+        Hints hints = new Hints(new RenderingHints(Hints.REPOSITORY, repository));
+        GridCoverage2DReader newReader = (format.gridFormat()).getReader(file, hints);
+        store.setURL(newFile);
+        CatalogBuilder builder = new CatalogBuilder(catalog);
+        builder.setStore(store);
+        builder.setWorkspace(store.getWorkspace());
+        CoverageInfo coverage;
+        try {
+            coverage = builder.buildCoverage(newReader, Collections.emptyMap());
+        } catch (Exception e) {
+            throw new IOException(
+                    "Exception occurred while configuring the coverage during the replace: ", e);
+        }
+
+        // Updating the old coverage in the catalog with the properties of the imported one
+        OwsUtils.copy(coverage, originalCoverage, CoverageInfo.class);
+
+        // Restore original names
+        originalCoverage.setName(name);
+        originalCoverage.setNativeName(nativeName);
+        catalog.save(originalCoverage);
+        catalog.save(store);
+
+        // Clean up the oldFile when not overwriting it
+        if (!newFile.equalsIgnoreCase(oldFile)) {
```

Comment:
Hum... maybe the "single file" thing has been taken a bit too far here.
AFAIK one can update also a WorldImage case, made of several files, and also the "single file" GeoTIFF might have external overviews, CRS forcing prjs, maybe mask files, and so on.

I think the could should work in terms of "set of files before" and "set of files after", where the set of files can be gathered using ``GridCoverage2DReader.getInfo(coverage)``.